src/Adapter/MSTest.TestAdapter/Execution/TestMethodRunner.cs

```diff
@@ -128,16 +129,187 @@ internal UnitTestResult[] Execute()
             result[result.Length - 1] = newResult;
 #pragma warning restore IDE0056 // Use index operator
         }
+
+        if (_testMethodInfo.Parent.ClassAttribute is STATestClassAttribute
```

Comment:
This is changing the logic of the method. Before, when a problem was thrown from assembly init, the rest of the code wasn't called while now it is.

Reply:
no,i have a boolean to check if assinit failed it won't enter the code only the cleanups as it was out this func